core/src/test/scala/org/apache/spark/sql/store/ColumnTableTest.scala

```diff
@@ -106,6 +106,37 @@ class ColumnTableTest
     println("Successful")
   }
 
+
+  test("Test SNAP-947") {
+    val table = "APP.TEST_TABLE"
+
+    snc.sql (s"drop table if exists $table")
+
+    // check that default concurrency checks is set to false for column table.
+    snc.sql(s"create table $table (col1 int) using column" )
+
+    assert (Misc.getRegionForTable(table , true).getAttributes.getConcurrencyChecksEnabled == false)
+
+    snc.dropTable(table)
+
+    // check that default concurrency checks setting is not modified.
+
+    snc.sql(s"create table $table (col1 int) using row options(PERSISTENT 'SYNCHRONOUS')" )
+
+    assert (Misc.getRegionForTable(table , true).getAttributes.getConcurrencyChecksEnabled == false)
+
+    snc.dropTable(table)
+
+    snc.sql(s"create table $table (col1 int) using row " +
+        s"options(PERSISTENT 'SYNCHRONOUS' , PARTITION_BY 'COL1')" )
+
+    assert (Misc.getRegionForTable(table , true).getAttributes.getConcurrencyChecksEnabled == true)
+
+    snc.dropTable(table)
```

Comment:
Test is fine. Please add a dunit test with persistent and eviction configured and do some restarts of vms and check too. Add a test for split mode too. Just for sanity it will be good. If something else will be there related to restarts it will be caught in the unit testing itself.